### PR TITLE
i#4953: Add missing KSTOP in trace building.

### PR DESCRIPTION
**Pull request by abhinav92003**
Adds a missing KSTOP in monitor_cache_enter for the case when we abort
trace building.

This was causing an assert failure in core/stats.c:361 'stop not matching TOS'
in the code_api,opt_memory|client.events test, on Ubuntu 18.04 in Github CI.

Issue: #4953

### CODE REVIEW

**Comment by abhinav92003** [comment]
Remaining failures after this fix:
```
2021-09-19T05:42:16.3077225Z debug-internal-32: 346 tests passed, **** 10 tests failed, of which 3 were flaky, but ignoring 2 for i#2941: ****
2021-09-19T05:42:16.3077960Z 	code_api|common.decode 
2021-09-19T05:42:16.3078696Z 	(ignore: i#2941) 	code_api|common.decode-stress 
2021-09-19T05:42:16.3079488Z 	(ignore: i#2941) 	code_api|linux.thread-reset 
2021-09-19T05:42:16.3080075Z 	code_api|linux.syscall_pwait 
2021-09-19T05:42:16.3080848Z 	code_api|tool.drcpusim.cpuid-Klamath 
2021-09-19T05:42:16.3081767Z 	code_api|tool.drcpusim.cpuid-Deschutes 
2021-09-19T05:42:16.3082662Z 	code_api|tool.histogram.offline 
2021-09-19T05:42:16.3083270Z 	code_api|api.detach_spawn_stress_FLAKY 
2021-09-19T05:42:16.3083857Z 	code_api|api.static_maps_mixup_novars_FLAKY 
2021-09-19T05:42:16.3084510Z 	code_api|api.static_sideline_FLAKY 
2021-09-19T05:42:16.3085373Z release-external-32: build successful; no tests for this build
```
https://github.com/DynamoRIO/dynamorio/pull/5108/checks?check_run_id=3643465970

File: core/monitor.c [hunk 1]
```diff
@@ -1991,6 +1991,7 @@ monitor_cache_enter(dcontext_t *dcontext, fragment_t *f)
                      * to last_copy, must re-clear!
                      */
                     md->last_fragment = NULL;
+                    KSTOP(trace_building);
```
Comment (abhinav92003) [inline core/monitor.c:1994]: @derekbruening  Not sure if it is worth tracking down why this path is being hit now. Let me know.

Comment (abhinav92003) [reply core/monitor.c:1994]: Weirdly this failure didn't show up on Github Actions for Ubuntu 18.04 #5072. But it did on my local workstation, so good to fix it anyway.

Comment (derekbruening) [reply core/monitor.c:1994]: Not worth further work.

**Review by derekbruening — APPROVED** [review]

**Pull request merged**